taxi/views.py

```diff
@@ -87,3 +88,58 @@
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+    template_name = "taxi/driver_form.html"
+    form_class = DriverCreateForm
+
+
+class DriverLicenseUpdateView(LoginRequiredMixin,
+                              UserPassesTestMixin,
+                              generic.UpdateView):
+    queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+    form_class = DriverLicenseUpdateForm
+    template_name = "taxi/license_form.html"
+
+    def get_success_url(self):
+        return reverse_lazy("taxi:driver-detail",
+                            kwargs={"pk": get_user(self.request).pk})
+
+    def test_func(self):
+        driver = self.get_object()
+        return self.request.user.id == driver.id
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["user"] = self.request.user
+        return context
+
+    def handle_no_permission(self):
+        return render(self.request, "taxi/license_form.html", {
+            "error_message": "You are not allowed to "
+                             "update another driver's "
+                             "license number!",
+        })
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+    template_name = "taxi/driver_confirm_delete.html"
+
+
+def driver_delete_from_car(request, driver_pk: int, car_pk: int):
+    driver = get_object_or_404(Driver, pk=driver_pk)
+    car = get_object_or_404(Car, pk=car_pk)
+    car.drivers.remove(driver)
```

Comment:
Consider checking if the driver is actually assigned to the car before attempting to remove them. This can prevent unnecessary database operations and potential errors.